src/GitHub.TeamFoundation.14/Connect/GitHubConnectSection.cs

```diff
@@ -344,22 +348,18 @@ public void Login()
         public bool OpenRepository()
         {
             var old = Repositories.FirstOrDefault(x => x.Equals(Holder.ActiveRepo));
-            // open the solution selection dialog when the user wants to switch to a different repo
-            // since there's no other way of changing the source control context in VS
             if (!Equals(SelectedRepository, old))
             {
-                if (ErrorHandler.Succeeded(ServiceProvider.GetSolution().OpenSolutionViaDlg(SelectedRepository.LocalPath, 1)))
+                var opened = vsServices.TryOpenRepository(SelectedRepository.LocalPath);
+                if(!opened)
```

Comment:
If both ways of opening a repository fail, `SelectedRepository` needs to be set back to the previous selection so that the UI reflects what's happened:

`SelectedRepository = old;`